Objects/dictobject.c

```diff
@@ -6692,8 +6701,26 @@ _PyObject_StoreInstanceAttribute(PyObject *obj, PyDictValues *values,
     assert(Py_TYPE(obj)->tp_flags & Py_TPFLAGS_MANAGED_DICT);
     Py_ssize_t ix = DKIX_EMPTY;
     if (PyUnicode_CheckExact(name)) {
-        LOCK_KEYS(keys);
+#ifdef Py_GIL_DISABLED
+        Py_hash_t hash = unicode_get_hash(name);
+        if (hash == -1) {
+            hash = PyUnicode_Type.tp_hash(name);
+            if (hash == -1) {
+                PyErr_Clear();
+                return DKIX_EMPTY;
```

Comment:
I think this should just be `return -1` without the `PyErr_Clear()`.

Reply:
That was a dumb copy paste... but actually this will never happen, on a collision with -1 it returns -2.  I'll add an assert.